### DISCUSSION TITLE
Allow values to display in Search UI attributes dropdown

### DISCUSSION THREAD

**Discussion by campmlc** [ArctosDB/arctos#8053] [Ideas]
This request is to have any controlled code table attribute values appear in the dropdown in the search menu when the value field is selected.  We need this for the examined for and detected code table values, at a minimum, but it should work for any code-table controlled value.

Currently, we see this in search:
![image](https://github.com/ArctosDB/arctos/assets/14808196/75ed3c90-202d-41ca-8830-d64d4071de75)

If we know what to start typing in the values field, and IF what we type matches something in the code table, we see a potential match show up that we can select.
BUT, if we type something that is not in the code table, there is no way to know that the negative results we get are because the option doesn't exist, or because there are no records. For example, if I type "cestodes" in the "examined for" or "detected" attribute value field, I get no records. But I happen to know that is because this option doesn't currently exist in the code table, not because we have no records with these data.

I can force the dropdown to show all options by using % in the value field. But that is not obvious to users. I would like the full dropdown menu to show all the options when I click on the attribute value field, as below, but without my having to enter %;

![image](https://github.com/ArctosDB/arctos/assets/14808196/a134765f-6ef6-4ca9-bf6c-ae1e01be0b01)

**Comment by jldunnum** [comment]
Agreed we need this, it will get a bit unwieldy but  is necessary to know what the possible options are

______________________________________________________________
Jonathan L. Dunnum Ph.D. (he, him, his)
Senior Collection Manager
Division of Mammals, Museum of Southwestern Biology
University of New Mexico
Albuquerque, NM 87131
(505) 277-9262
Fax (505) 277-1351

Chair, Systematic Collections Committee, American Society of Mammalogists
Latin American Fellowship Committee, ASM

MSB Mammals website: http://www.msb.unm.edu/mammals/index.html
Facebook: http://www.facebook.com/MSBDivisionofMammals

Shipping Address:
Museum of Southwestern Biology
Division of Mammals
University of New Mexico
CERIA Bldg 83, Room 204
Albuquerque, NM 87131


From: Mariel ***@***.***>
Sent: Thursday, September 21, 2023 11:30 AM
To: ***@***.***>
Cc: ***@***.***>
Subject: [ArctosDB/arctos] Allow values to display in Search UI attributes dropdown (Issue #6759)

  [EXTERNAL]

This request is to have any controlled code table attribute values appear in the dropdown in the search menu when the value field is selected. We need this for the examined for and detected code table values, at a minimum, but it should work for any code-table controlled value.

Currently, we see this in search:
[image]<https://user-images.githubusercontent.com/14808196/269703062-75ed3c90-202d-41ca-8830-d64d4071de75.png>

If we know what to start typing in the values field, and IF what we type matches something in the code table, we see a potential match show up that we can select.
BUT, if we type something that is not in the code table, there is no way to know that the negative results we get are because the option doesn't exist, or because there are no records. For example, if I type "cestodes" in the "examined for" or "detected" attribute value field, I get no records. But I happen to know that is because this option doesn't currently exist in the code table, not because we have no records with these data.

I can force the dropdown to show all options by using % in the value field. But that is not obvious to users. I would like the full dropdown menu to show all the options when I click on the attribute value field, as below, but without my having to enter %;

[image]<https://user-images.githubusercontent.com/14808196/269704912-a134765f-6ef6-4ca9-bf6c-ae1e01be0b01.png>

**Comment by campmlc** [comment]
I don't understand why this was moved to discussion, and what the mystery is. The request is clear. What are the problems preventing this from being actionable?

**Comment by dustymc** [comment]
This just isn't compatible with the current libraries in any way I can identify.

We are still using https://github.com/ArctosDB/arctos/issues/5553, so identifying some replacement (or different approach, or alternative, or WHATEVER) would be most useful, and potentially applicable to much more than the one UI.
